File: sample-conference-videochat/sample-conference-videochat/Components/Users/Views/CheckView/CheckView.m
//
//  CheckView.m
//  LoginComponent
//
//  Created by Andrey Ivanov on 03/06/16.
//  Copyright © 2016 Quickblox. All rights reserved.
//

#import "CheckView.h"

@interface CheckView()

@property (strong, nonatomic) UIImageView *imageView;

@end

@implementation CheckView

- (UIImage *)checkboxNormalImage {
    
    static UIImage *_qm_checkbox_normal = nil;
    
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        _qm_checkbox_normal = [UIImage imageNamed:@"checkbox-normal"];
    });
    
    return _qm_checkbox_normal;
}

- (UIImage *)checkboxPressedImage{
    
    static UIImage *_qm_checkbox_pressed = nil;
    
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        
        _qm_checkbox_pressed = [UIImage imageNamed:@"checkbox-pressed"];
    });

    return _qm_checkbox_pressed;
}

- (void)awakeFromNib {
    [super awakeFromNib];
    
    _imageView = [[UIImageView alloc] initWithImage:[self checkboxNormalImage]];
    _imageView.frame = self.bounds;
    [self addSubview:_imageView];
}

- (void)layoutSubviews {
    
    [super layoutSubviews];
    
    _imageView.frame = self.bounds;
}

- (void)setCheck:(BOOL)check {
    
    if (_check != check) {
        _check = check;
        
        self.imageView.image = check ? [self checkboxPressedImage] : [self checkboxNormalImage];
    }
}

@end

File: sample-conference-videochat-swift/sample-conference-videochat-swift/sample-conference-videochat-swift/Components/Users/Views/CheckView/CheckView.swift
//
//  CheckView.swift
//  sample-conference-videochat-swift
//
//  Created by Vladimir Nybozhinsky on 17.10.2018.
//  Copyright © 2018 QuickBlox. All rights reserved.
//

import UIKit

class CheckView: UIView {
    
    let checkboxNormalImage = UIImage(named: "checkbox-normal")
    let checkboxPressedImage = UIImage(named: "checkbox-pressed")
    private var imageView: UIImageView!
    
    var check: Bool? {
        didSet {
            setCheck(check!)
        }
    }
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        imageView = UIImageView(image: checkboxNormalImage)
        imageView.frame = bounds
        addSubview(imageView)
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        imageView.frame = bounds
    }
    
    func setCheck(_ check: Bool) {
        imageView.image = check ? checkboxPressedImage : checkboxNormalImage
    }
}
